### DISCUSSION TITLE
Feature: Export + Search The Reading History (URI+TITLE ==> text/excel file)

### DISCUSSION THREAD

**Discussion by MohamedHakem** [dailydotdev/daily#398] [Feature Requests]
### Hi!
I was going to suggest a Reading history, but 2 days ago I found it already present in the new design! 
That's exactly what I needed! This is amazing! Thank you! * Btw, the new design is awesome.

Now, the first thing that comes to my mind when I go to the `reading history`  is 
to search for something I've read, and whether there's an Export button somewhere on the page!

### So, I thought I'd just suggest both of them to you: 
1- `Export`:  export all the reading history in ULI/Title format as a text, excel file or something similar. (maybe similar to OneTab export, screenshot attached below)

2- `Search`:  It would be great if I can search for some word that I remember from an article I have read (through daily.dev), 
and the search engine searches for that word in the Titles or URL of my reading history.

Also, it would be great to have the same search functionality on the main page! **Filters and Search!** 

Lastly, thank you!  I'm using Daily.dev daily since the first time I saw it, and it helps me with a really annoying problem (searching for new quality articles/insights).  It's amazing. 

![onetab-ext-export](https://user-images.githubusercontent.com/29775958/147415260-66868bae-39ee-4483-8a84-1d3321c2f31a.JPG)

**Comment by idoshamun** [comment]
Hi @MohamedHakem,

Searching in the reading history is definitely on our roadmap.
As for the export functionality, I'll raise it in our upcoming product meeting and let you know :)

**Comment by nimrodkra** [comment]
> 1- `Export`: export all the reading history in ULI/Title format as a text, excel file or something similar. (maybe similar to OneTab export, screenshot attached below)

Hi @MohamedHakem,
Can you please give more background as to why you want to export your reading history? How is it going to help you?

**Comment by MohamedHakem** [comment]
@nimrodkra 
I want to export my reading history so I can 
- Filter out any irrelevant/low-quality articles and then I could:
- Sort it and play with it programmatically to find some useful insights about my reading habits, etc.
- Share it with others.
- Use the results on my blog as a recommendation/guide page (potentially interesting/useful readings for others).
- Save it on my drive for later.  
- Probably will have a few other ideas by the time I start filtering/sorting it.

The Title and link (similar to the above screenshot) would be enough to filter/sort/search/categorize/count them, find some insights, display these insights somehow, and share/recommend useful articles/guides to others.
It's similar to the Dev Card minus the limitations. 

---
Another suggestion (if exporting isn't a possibility for the near future) would be, adding a small copy button/icon beside the X icon on each item in the reading history that allows the user to copy the `[Title+Link]` as text for example, instead of the user doing the following:
- Click the title, the link opens in a new tab
- copy the link
- paste somewhere
- copy the title of the article
- paste somewhere
- close the tab

And of course, either way: **Thank you**.

**Reply by nimrodkra** [reply]
@MohamedHakem Thank you very much for providing such a rich context on your request. It's helpful!
As @idoshamun we'll discuss it in our upcoming product meeting :)

**Comment by nimrodkra** [comment]
Hi @MohamedHakem I wanted to let you know that we shppied the Search Reading History feature. Thank you for the feedback and for helping us make daily.dev better! You can give it a try yourself here: https://app.daily.dev/history

![Screen Shot 2022-02-09 at 16 17 27](https://user-images.githubusercontent.com/41470823/153219261-67a1ac20-a31b-48da-9ff7-31526d92437d.png)
